aux1.c

```diff
@@ -512,10 +512,21 @@ char *fromstring;
   dataprint();
 }
 
+/* get save filename */
+void gen_save_fname(save_file)
+char *save_file;
+{
+  strcpy(save_file, SAVEDIR);
+  strcat(save_file, Player.name);
+  strcat(save_file, ".sav");
```

Comment:
This could probably be written as:
```c
void gen_save_fname( save_file, max_len )
char *save_file;
int max_len;
{
  snprintf( save_file, max_len, "%s%s.sav", SAVEDIR, Player.name );
}
```
Would change calls slightly.

Could also be written as
```c
void gen_save_fname( save_file )
char *save_file;
{
  sprintf( save_file, "%s%s.sav", SAVEDIR, Player.name );
}
```
which would not require changing the function signature by adding a length input parameter